app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/SessionsAdapter.kt

```diff
@@ -0,0 +1,232 @@
+        resetTextStyle(viewHolder.room, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.duration, R.style.ScheduleListSecondary)
+        viewHolder.withoutVideoRecording.setImageResource(R.drawable.ic_without_video_recording)
+    }
+
+    protected open fun resetTextStyle(textView: TextView, style: Int) {
+        textView.setTextAppearance(context, style)
+    }
+
+    protected abstract fun setItemContent(position: Int, viewHolder: ViewHolder)
+
+    private fun setSeparatorContent(position: Int, viewHolderSeparator: ViewHolderSeparator?) {
+        viewHolderSeparator!!.text!!.text = mSeparatorStrings!![mMapper!![position]]
+    }
+
+    override fun getViewTypeCount(): Int = NUM_VIEW_TYPES
+
+    override fun areAllItemsEnabled(): Boolean = false
+
+    override fun isEnabled(position: Int): Boolean = !mSeparatorsSet!!.contains(position)
+
+    override fun getItemViewType(position: Int): Int =
+        if (mSeparatorsSet!!.contains(position))
+            TYPE_SEPARATOR
+        else
+            TYPE_ITEM
+
+    override fun getCount(): Int {
+        var count = 0
+
+        if (list != null) {
+            count += list.size
+        }
+
+        if (mSeparatorsSet != null) {
+            count += mSeparatorsSet!!.size
+        }
+
+        return count
+    }
+
+    private fun initMapper() {
+        mSeparatorsSet = TreeSet()
+        mSeparatorStrings = ArrayList()
+        mMapper = ArrayList()
+
+        var day: Int
+        var lastDay = 0
+        var sepCount = 0
+
+        if (list == null) {
+            return
+        }
+
+        val daySeparator = context!!.getString(R.string.day_separator)
+        for (index in list.indices) {
+            val l = list[index]
+            day = l.day
+
+            if (day != lastDay) {
+                lastDay = day
+                if (numDays > 1) {
+                    val formattedDate = newInstance(useDeviceTimeZone).getFormattedDate(l.dateUTC, l.timeZoneOffset)
+                    val dayDateSeparator = String.format(daySeparator, day, formattedDate)
+
+                    mSeparatorStrings!!.add(dayDateSeparator)
+                    mSeparatorsSet!!.add(index + sepCount)
+                    mMapper!!.add(sepCount)
+
+                    sepCount++
+                }
+            }
+
+            mMapper!!.add(index)
+        }
+    }
+
+    override fun notifyDataSetChanged() {
+        super.notifyDataSetChanged()
+        initMapper()
+    }
+
+    data class ViewHolder(
+        var title: TextView,
+        var subtitle: TextView,
+        var speakers: TextView,
+        var lang: TextView,
+        var day: TextView,
+        var time: TextView,
+        var room: TextView,
+        var duration: TextView,
+        var noVideo: ImageView,
+        var video: ImageView,
+        var withoutVideoRecording: ImageView
+    )
+
+    internal class ViewHolderSeparator {
+        var text: TextView? = null
+    }
+
+    init {
```

Comment:
Please move the `init` function up below the class properties. Please also check if the properties need to be nullable or can be `lateinit` and non-null because they are initialized right after the class instance is constructed.